Added a way to unregister keybindings for a command

JonasHelming opened the pull request:
fixes: #8209
Signed-off-by: Jonas Helming <jhelming@eclipsesource.com>

#### What it does
It adds the capability to remove all keybindings for a specific command. also adds docu to the three available functions.

#### How to test
see included unit tests or unregister the keybindings of any command via API

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

JonasHelming commented:
This is based on the comment: https://github.com/eclipse-theia/theia/issues/8209#issuecomment-661686777

JonasHelming commented:
This is a duplicate of https://github.com/eclipse-theia/theia/pull/8257 but with unit tests

JonasHelming commented:
No clue whats wron with the build, but I have the same failure on latest master

kittaakos commented on packages/core/src/browser/keybinding.spec.ts:
```diff
@@ -173,6 +173,109 @@ describe('keybindings', () => {
+        keybindingRegistry.unregisterKeybinding(TEST_COMMAND2);
+
+        const bindings = keybindingRegistry.getKeybindingsForCommand(TEST_COMMAND2.id);
+        if (bindings) {
```
Why not `expect`? Same the other `if`.

JonasHelming replied:
Not really the thing we test here, but I will adapt that.

kittaakos commented on packages/core/src/browser/keybinding.spec.ts:
```diff
@@ -173,6 +173,109 @@ describe('keybindings', () => {
+
+        const bindings = keybindingRegistry.getKeybindingsForCommand(TEST_COMMAND2.id);
+        if (bindings) {
+            expect(bindings.length).to.be.equal(0);
```
Let's add an `expect` before unregister.

JonasHelming replied:
Good point! I was expecting that there are test cases that validate the basic registration, but apparently there are none. I adapted that.

kittaakos commented on packages/core/src/browser/keybinding.spec.ts:
```diff
@@ -173,6 +173,109 @@ describe('keybindings', () => {
+        if (bindings) {
+            bindings.forEach(
+                (value: Keybinding) => {
+                    expect(value.command === keybinding.command && value.keybinding === keybinding.keybinding).to.be.false;
```
`value.keybinding === keybinding.keybinding` => https://github.com/eclipse-theia/theia/issues/8170

JonasHelming replied:
Let us fix that bug separatly, I can prepare a contribution

kittaakos commented on packages/core/src/browser/keybinding.spec.ts:
```diff
@@ -173,6 +173,109 @@ describe('keybindings', () => {
+
+        keybindingRegistry.registerKeybinding(keybinding);
+        keybindingRegistry.registerKeybinding(keybinding2);
+
```
I think an `expect` is missing here. These tests also pass if register/unregister does not work at all.

JonasHelming replied:
Same as above

JonasHelming commented:
- I extracted a function for the check that a keybinding is registered (much easier to read now)
- I added the requested checks that a keybinding is registered before unregistering
- I added one simple test case just for the registration
- I removed the "number checks" cause every keybinding is checked specifically now

JonasHelming commented:
I resolved the conflicts

kittaakos commented on packages/core/src/browser/keybinding.ts:
```diff
@@ -184,28 +184,36 @@ export class KeybindingRegistry {
 
     /**
-     * Unregister keybinding from the registry
+     * Unregister keybindings from the registry using the key of the given keybinging
```
There is a typo here: `keybinging`.

kittaakos reviewed (approved):
Thank you for taking care of this PR, @JonasHelming. Updated jsdocs, tests, convenient API 👍
